### ISSUE TITLE
v12 - Cannot set default value for `Umbraco.DropDown.Flexible` in block content using `SendingContentNotification`

### ISSUE THREAD

**Issue by timeverts** [umbraco/Umbraco-CMS#14914]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

12.2.0

### Bug summary

I've defined an `Umbraco.DropDown.Flexible` property editor inside of an element type that I am using as the content model for a Block List block.  I have attempted to set the default value for the drop down property editor using a `SendingContentNotification` handler, using code similar to the example in the [Umbraco documentation](https://docs.umbraco.com/umbraco-cms/reference/notifications/editormodel-notifications#usage). 

However, setting the `Value` of the property editor within the `SendingContentNotification` handler has no impact for the drop down within the block content model. When the block is displayed, the drop down is initialised with the empty/blank option selected, rather than the value specified in the `SendingContentNotification` handler.

Worth noting is that the same code _does_ set the default value correctly when the drop down property is used in a document type, rather than a block content model. So I suspect this is isolated to the initialisation of block content model interfaces.

This might be related to a similar problem described in #14378 and potentially could have been the actual scenario in #13169.

### Steps to reproduce

1. Create an element type with an `Umbraco.DropDown.Flexible` property editor.
For example, a 'Heading' `[heading]` element type:

![image](https://github.com/umbraco/Umbraco-CMS/assets/15150632/548016fb-21a0-4de2-be1b-a9d6cbe62f68)

The heading 'Type' drop down property is configured like below:
![image](https://github.com/umbraco/Umbraco-CMS/assets/15150632/d27db208-2698-4643-a5b4-9003d98b10c0)

2. Create a Block List field that includes the 'Heading' element type as a Content Model for a block:
![image](https://github.com/umbraco/Umbraco-CMS/assets/15150632/9c2a2dc2-9772-47dc-acbd-5829f9b05a4a)

3. Create a `SendingContentNotification` handler to set the default value for the `type` property of the `heading` block:
```
public class EditorSendingContentNotificationHandler : INotificationHandler<SendingContentNotification>
{
    public void Handle(SendingContentNotification notification)
    {
        if (notification.Content.ContentTypeAlias.Equals("heading"))
        {
            foreach (var variant in notification.Content.Variants)
            {
                if (variant.State == ContentSavedState.NotCreated)
                {
                    var headingTypeProperty = variant.Tabs.SelectMany(f => f.Properties)
                        .FirstOrDefault(f => f.Alias.InvariantEquals("type"));

                    if (headingTypeProperty is not null)
                    {
                        // set default value of the heading type property to 'Heading 2'
                        headingTypeProperty.Value = new[] { "Heading 2" };

                        // change the description of the heading type property to verify the handler is being executed
                        headingTypeProperty.Description = "Description set from notification";
                    }
                }
            }
        }
    }
}
```
4. Register the handler via a composer:
```
public class NotificationHandlerComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        builder.AddNotificationHandler<SendingContentNotification, EditorSendingContentNotificationHandler>();
    }
}
```

### Expected result / actual result

The expected result is that when a new 'Heading' block is created, the default value of the 'Type' drop down will be 'Heading 2'.  Instead, the blank/empty option is selected.

![image](https://github.com/umbraco/Umbraco-CMS/assets/15150632/7cc69551-14f6-4e23-9360-b8678382f3b1)

Note that you can see the drop down property's description has been updated from the `SendingContentNotification` handler. 

Using Visual Studio debugger I also verified that the `Value` property of the editor was set correctly to a `{string[1]}`. So there seems to be something wrong in the initalisation of the block list content view for an `Umbraco.DropDown.Flexible`. 

As mentioned earlier, putting the 'Type' property directly in a Document Type sets the default value correctly from the  `SendingContentNotification` handler, so it's definately isolated to block content model views.

**Comment by sonikachandel13** [comment]
hi @timeverts 
 
What is the resolution of the above issue. Please help me

**Comment by timeverts** [comment]
@sonikachandel13 , @Rockerby has kindly done some work to address this and submitted a pull request. So hopefully it will be ready in the coming weeks.

**Comment by sonikachandel13** [comment]
hi @timeverts,

Thanks for the update. Really appreciate your response .Will the fix be applicable for Umbraco version 8.00 ..

**Comment by timeverts** [comment]
@sonikachandel13, I don't believe this will be applicable for Umbraco 8. Only 10+.
